DDCore/src/MatrixHelpers.cpp

```diff
@@ -112,6 +112,19 @@ Transform3D dd4hep::detail::matrix::_transform(const TGeoMatrix* matrix)    {
                      0e0,0e0,0e0,t[1]*MM_2_CM,
                      0e0,0e0,0e0,t[2]*MM_2_CM);
 }
+// add another implementation that takes const reference
+Transform3D dd4hep::detail::matrix::_transform(const TGeoMatrix& matrix)    {
+  const Double_t* t = matrix.GetTranslation();
+  if ( matrix.IsRotation() )  {
+    const Double_t* r = matrix.GetRotationMatrix();
+    return Transform3D(r[0],r[1],r[2],t[0]*MM_2_CM,
```

Comment:
There are a lot of conversions here. It looks like the MM_2_CM and RAD_2_DEGREE assume that the input units are mm and rad. Should it be a user who defines which units to use and if they are not the DD4hep default, convert them?

Reply:
The user should always use DD4hep units, when addressing/talking to DD4hep. There some exceptions to this rule, e.g. when interfaces of underlying classes are exposed, there you should use or expect ROOT or Geant4 conventions.